packages/xml-for-vscode/src/test/suite/extension.test.ts

```diff
@@ -0,0 +1,127 @@
+  <body>Don't forget me this weekend!</body>
+</note>`;
+    await setContent(content);
+    await testDiagnostics([
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message:
+          "unexpected character: ->!<- at offset: 5, skipped 1 characters.",
+        range: toRange(0, 5, 0, 6),
+        source: SYNTAX_ERROR_MSG
+      }
+    ]);
+  });
+
+  it("Test diagnostic text - parsing errors", async function() {
+    let content = `<note>
+  <to a=>Tove</to>
+  <from>Jani</from>
+  <heading>Reminder</heading>
+  <body>Don't forget me this weekend!</body>
+</note`;
+    await setContent(content);
+    await testDiagnostics([
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(1, 8, 1, 8),
+        source: SYNTAX_ERROR_MSG
+      },
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> CLOSE <-- but found --> '' <--",
+        range: toRange(5, null, 0, 0),
+        source: SYNTAX_ERROR_MSG
+      }
+    ]);
+  });
+
+  it("Test diagnostic text - Multiple errors", async function() {
+    let content = `<note>
+  <to a=>Tove</to>
+  <from a=>Jani</from>
+  <heading a=>Reminder</heading>
+  <body>Don't forget me this weekend!</body>
+</note>`;
+    await setContent(content);
+    await testDiagnostics([
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(1, 8, 1, 8),
+        source: SYNTAX_ERROR_MSG
+      },
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(2, 10, 2, 10),
+        source: SYNTAX_ERROR_MSG
+      },
+      {
+        severity: vscode.DiagnosticSeverity.Error,
+        message: "Expecting token of type --> STRING <-- but found --> '>' <--",
+        range: toRange(3, 13, 3, 13),
+        source: SYNTAX_ERROR_MSG
+      }
+    ]);
+  });
+
+  function toRange(sLine: number, sChar: number, eLine: number, eChar: number) {
+    const start = new vscode.Position(sLine, sChar);
+    const end = new vscode.Position(eLine, eChar);
+    return new vscode.Range(start, end);
+  }
+
+  async function setContent(content: string) {
+    await vscode.workspace.openTextDocument(docUri);
+    await fs.writeFile(docPath, content);
+    await sleep(2000);
+    await vscode.window.showTextDocument(docUri);
+  }
+
+  async function testDiagnostics(expectedDiagnostics: vscode.Diagnostic[]) {
+    let actualDiagnostics: vscode.Diagnostic[] = vscode.languages.getDiagnostics(
+      docUri
+    );
+    console.log(JSON.stringify(actualDiagnostics));
+    console.log(JSON.stringify(expectedDiagnostics));
+    expect(actualDiagnostics.length).to.equal(expectedDiagnostics.length);
+
+    expectedDiagnostics.forEach((expectedDiagnostic, i) => {
```

Comment:
- try to use deep.equals or include.members instead of iterating over expected diagnostics.
- try to avoid assumptions on order (include.members > deep.equals)